### PR TITLE
Validate character name before attempting creation.

### PR DESCRIPTION
**Pull request by sysrage**

### CODE REVIEW

File: shared/widgets/CharacterCreation/index.tsx [hunk 1]
```diff
@@ -85,36 +85,45 @@ class CharacterCreation extends React.Component<CharacterCreationProps, any> {
...
-          acc[cur.name] = cur.allocatedPoints;
-        } else {
-          acc[cur.name] += cur.allocatedPoints;
-        }
-        return acc;
-      }),
+    // validate name
+    const modelName = (this.refs['name-input'] as any).value.trim();
+    let errors: any = [];
+    if (modelName.length < 2 || modelName.length > 20) errors.push('A character name must be between 2 and 20 characters in length.');
```
Comment (codecorsair) [inline shared/widgets/CharacterCreation/index.tsx]: The length of 2 is after normalization. So, the name ( O' ) would pass your validation, but not the server.  ( O'B ) would pass as it has a min of 2 after normalization.

Additionally, I think I neglected to mention this in the trello card, you can not have consecutive special characters.  ( O``B ) is invalid, ( O--B ) is invalid  ( O-B' ) is valid.

Lastly, the first character must be a letter.  ( -OB ) is invalid  ( O-B ) is valid

**Comment by sysrage** [comment]
I believe all listed concerns have now been addressed.

File: shared/widgets/CharacterCreation/index.tsx [hunk 2]
```diff
@@ -85,36 +85,48 @@ class CharacterCreation extends React.Component<CharacterCreationProps, any> {
...
-        } else {
-          acc[cur.name] += cur.allocatedPoints;
-        }
-        return acc;
-      }),
+    // validate name
+    const modelName = (this.refs['name-input'] as any).value.trim();
+    const normalName = modelName.replace(/[^a-zA-Z]/g, '').toLowerCase();
+    let errors: any = [];
+    if (normalName.length < 2 || normalName.length > 20) errors.push('A character name must be between 2 and 20 characters in length (not including hyphens or apostrophes).');
```
Comment (codecorsair) [inline shared/widgets/CharacterCreation/index.tsx]: max length of 20 is before normalization.  so  if (normalName.length < 2 || modelName > 20)...

File: shared/widgets/CharacterCreation/index.tsx [hunk 3]
```diff
@@ -85,36 +85,48 @@ class CharacterCreation extends React.Component<CharacterCreationProps, any> {
...
+    if (modelName.search(/^[a-zA-Z]/) === -1) errors.push('A character name must begin with a letter.');
+    if (modelName.search(/[\-']{2,}/) > -1) errors.push('A character name must not contain two or more consecutive hyphens (-) or apostrophes (\').');
```
Comment (codecorsair) [inline shared/widgets/CharacterCreation/index.tsx]: This should fail if there are two consecutive punctuation at all. "J-B-`-`-`".search(/[\-']{2,}/) === -1 which should be an invalid name but will pass.

**Comment by sysrage** [comment]
Good now?

**Comment by codecorsair** [comment]
:+1: Looks good.

**Pull request merged**